### PR TITLE
Terraform tester env & new path structure

### PR DESCRIPTION
**Pull request by evaldofelipe**
This PR create a new vm structure to test a image was created by packer, and has a new path structure to grow the project in the future.

### CODE REVIEW

File: terraform/providers/azure/images-tester/tester-environment.tf [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+# # Image from packer reference to build a new VM
+# data "azurerm_image" "builder_image" {
+#   name                = "${var.packer_image_name}"
+#   resource_group_name = "${var.packer_resource_group_name}"
+# }
+
+
+resource "azurerm_image" "builder_image" {
+  name     = "${var.packer_image_name}"
```
Comment (OctavioBR) [inline terraform/providers/azure/images-tester/tester-environment.tf]: Where this `packer_image_name` comes from?

File: Makefile [hunk 2]
```diff
@@ -15,15 +26,26 @@ base-docker-run = docker run \
...
+		exit 1; \
+	fi
 
 terraform-docker-run = $(base-docker-run) \
-	-w  /packer-images/terraform \
-	-it default-image-azure
+	-w  /packer-images/$(provider-dir) \
+	-it packer-images
+
+terraform-docker-run = $(base-docker-run) \
```
Comment (OctavioBR) [inline Makefile]: This `terraform-docker-run` is duplicated

File: Makefile [hunk 3]
```diff
@@ -15,15 +26,26 @@ base-docker-run = docker run \
 	-e AZURE_SUBSCRIPTION_ID=$(AZURE_SUBSCRIPTION_ID) \
 	-e AZURE_TENANT=$(AZURE_TENANT_ID) \
 	-e AZURE_TENANT_ID=$(AZURE_TENANT_ID) \
-	-v $(shell pwd):/packer-images
+	-v $(shell pwd):/packer-images \
+	$(docker_ssh_opts) \
+
+guard-%:
```
Comment (OctavioBR) [inline Makefile:34]: Have you mentioned the necessity of exporting `env="something"` before running terraform make targets?

Comment (OctavioBR) [reply Makefile:34]: Looks like only `images-builder` and `images-tester` are valid options.  
@lborguetti why do we use GNU Make as our "task runner"?  
I really would like something nicer for reading parameters. Maybe write our CLI in bash/python/go. Don't know what people usually do 🤔

**Review by OctavioBR — CHANGES_REQUESTED** [review]
The only thing that I definitely consider needing a fix is the duplicated make variable `terraform-docker-run`.  
Other than that LGTM. I've reached you through slack to discuss the necessity of those 4 "single file folders" for `providers → azure → ...`

File: terraform/providers/azure/images-tester/tester-environment.tf (hunk 1 above)
Comment (evaldofelipe) [reply terraform/providers/azure/images-tester/tester-environment.tf]: This variable come from `variables.tf` file. This file is a syslink between variables from packer environment and the tester environment. But maybe this structure will change in the next commit

File: Makefile (hunk 2 above)
Comment (evaldofelipe) [reply Makefile]: Thanks!

File: Makefile (hunk 3 above)
Comment (evaldofelipe) [reply Makefile:34]: It's necessary declare `env=` to execute the targets, but it's not documented yet.

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
 
+provider-dir = terraform/providers/azure/$(env)
+
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Linux)
+	docker_ssh_opts =  -e SSH_AUTH_SOCK=$(SSH_AUTH_SOCK) \
+	-v $(SSH_AUTH_SOCK):$(SSH_AUTH_SOCK)
+endif
+ifeq ($(UNAME_S),Darwin)
+	docker_ssh_opts = -v $(HOME)/.ssh:/home/packer/.ssh:ro
```
Comment (lborguetti) [inline Makefile]: Have you ever had problems with this mount point? :thinking: 

I think you just need your ssh key ;)

File: Makefile (hunk 3 above)
Comment (lborguetti) [reply Makefile:34]: @OctavioBR about why we used GNU Make,  we just clone the environment used by other team. 

I like to use GNU Make for stuff like that, but we can switch to our CLI too.

**Review by OctavioBR — APPROVED** [review]
Great job 👌

File: Makefile [hunk 5]
```diff
@@ -15,15 +26,22 @@ base-docker-run = docker run \
...
+	@ if [ "${${*}}" = "" ]; then \
+		echo "Variable '$*' not set"; \
+		exit 1; \
+	fi
 
 terraform-docker-run = $(base-docker-run) \
-	-w  /packer-images/terraform \
-	-it default-image-azure
+	-w  /packer-images/$(provider-dir) \
+	-it packer-images
```
Comment (lborguetti) [inline Makefile]: We don't need pass `--interactive`  mode here.

File: Makefile [hunk 6]
```diff
@@ -15,15 +26,22 @@ base-docker-run = docker run \
...
 
 packer-docker-run = $(base-docker-run) \
-	-w  /packer-images/packer \
-	-it default-image-azure
+	-w  /packer-images/packer/builders/azure/image-ubuntu/ \
+	-it packer-images
```
Comment (lborguetti) [inline Makefile]: We don't need pass `--interactive`  mode here.

File: packer/builders/azure/image-ubuntu/provisioner.json [hunk 7]
```diff
@@ -16,24 +16,18 @@
     "image_offer": "UbuntuServer",
     "image_sku": "16.04-LTS",
 
-    "managed_image_resource_group_name": "rg-packer-azure",
+    "managed_image_resource_group_name": "packer-images-resource-group",
     "managed_image_name": "default-image-azure",
```
Comment (lborguetti) [inline packer/builders/azure/image-ubuntu/provisioner.json]: @evaldofelipe What do you think about refer Ubuntu here?

**Review by lborguetti — CHANGES_REQUESTED** [review]
@evaldofelipe :1st_place_medal:

File: packer/builders/azure/image-ubuntu/provisioner.json (hunk 7 above)
Comment (evaldofelipe) [reply packer/builders/azure/image-ubuntu/provisioner.json]: Agree.

File: Makefile (hunk 4 above)
Comment (evaldofelipe) [reply Makefile]: Great advice! Fixed.

File: Makefile (hunk 5 above)
Comment (evaldofelipe) [reply Makefile]: Fixed.

File: Makefile (hunk 6 above)
Comment (evaldofelipe) [reply Makefile]: Fixed.

**Review by lborguetti — APPROVED** [review]
LGTM!

**Pull request merged**